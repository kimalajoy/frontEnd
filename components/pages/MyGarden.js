import React, { Component } from 'react';
import { View, Text, ImageBackground, SafeAreaView, ScrollView} from 'react-native';

import GardenPlant from '../GardenPlant';
import styles from '../../styles/styles';
import dirtBackground from '../../assets/dirt.png';
import api from '../../api/plantAPI'

export default class MyGarden extends Component {
  state = {
    gardenPlants: [],
  }

  componentDidMount() {
    this.getAllGardenPlants()
  }

  componentDidUpdate() {
    this.getAllGardenPlants()
  }

  getAllGardenPlants = () => {
    api.getAllGardenPlants()
    .then(response => {
      this.setState({gardenPlants: response.data})
    })
    .catch(err => console.log(err))
  }

  render(){
    return (
      <SafeAreaView style={styles.myGardenContainer}>
        <ImageBackground
          style={styles.dirtBackground}
          source={dirtBackground}>
          <View style={styles.myGardenContainer}>
            <View style={styles.myGardenHeader}>
              <Text style={{color: 'white'}}>My Garden</Text>
            </View>
            <ScrollView >
              <View style={styles.myGarden}>
                {this.state.gardenPlants.map( plant => {
                  return(
                    <GardenPlant navigation={ this.props.navigation } info={plant} key={plant.id}/>
                  )
                })}
              </View>
            </ScrollView>
          </View>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}
