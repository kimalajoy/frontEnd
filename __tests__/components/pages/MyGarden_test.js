import MyGarden from '../../../components/pages/MyGarden';
import renderer from 'react-test-renderer';
import React from 'react';


describe("MyGarden Page", () => {
  let myGardenPage;

  beforeAll(() => {
    myGardenPage = renderer.create(<MyGarden />)
  })

  it('MyGarden Page does render', () => {
    expect(myGardenPage).toMatchSnapshot();
  })
})
