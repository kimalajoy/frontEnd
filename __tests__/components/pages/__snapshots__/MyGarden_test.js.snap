// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyGarden Page MyGarden Page does render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffefe",
      "color": "#fafafa",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    YOU MADE IT TO YOUR GARDEN!!
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Plant Profile Page
      </Text>
    </View>
  </View>
</View>
`;
